### PR TITLE
Doxygen groups for C headers.

### LINKED ISSUE #89: Generated C headers don't close Doxygen brace
e.g. https://github.com/henryhallam/libsbp/blob/ext_events/c/include/libsbp/ext_events.h line 22 opens a Doxygen defgroup but it's never closed.

### PR DESCRIPTION
**Pull request by mookerji**
Adds groups, packaging information, and correct struct handling for
the Doxygen-generated C documentation.

Closes #89.

/cc @henryhallam

### CODE REVIEW

File: c/docs/Doxyfile.in [hunk 1]
```diff
@@ -1522,25 +1522,19 @@ PERLMOD_MAKEVAR_PREFIX =
 # Configuration options related to the preprocessor
 #---------------------------------------------------------------------------
 
-# If the ENABLE_PREPROCESSING tag is set to YES (the default) Doxygen will
+# If the ENABLE_PREPROCESSING tag is set to YES (the default) Oxygen will
```
Comment (henryhallam) [inline c/docs/Doxyfile.in]: Oxygen?

File: c/docs/Doxyfile.in [hunk 2]
```diff
@@ -1848,3 +1842,9 @@ GENERATE_LEGEND        = YES
 # the various graphs.
 
 DOT_CLEANUP            = YES
+
+# See: http://doxygen.10944.n7.nabble.com/struct-attribute-packed-not-working-td2666.html
+ENABLE_PREPROCESSING   = YES 
+MACRO_EXPANSION        = YES 
+EXPAND_ONLY_PREDEF     = YES 
+PREDEFINED             = __attribute__(x)=
```
Comment (henryhallam) [inline c/docs/Doxyfile.in]: missing newline

**Comment by henryhallam** [comment]
lgtm other than comments

File: c/docs/Doxyfile.in (hunk 2 above)
Comment (mookerji) [inline c/docs/Doxyfile.in]: Done

**Pull request merged**